### ISSUE TITLE
no-invalid-require might be too strict in requiring FunctionExpression nodes as last arguments

### ISSUE THREAD

**Issue by platinumazure** [cvisco/eslint-plugin-requirejs#44]
The `no-invalid-require` rule has some comments about wanting to flag only obviously incorrect cases and not flagging expressions that could _evaluate_ to correct cases. However, the checks for FunctionExpression nodes (as last arguments of 2- or 3-argument forms of `require()`) are stricter than that, and are causing some problems in my application.

Here are the two examples I'm working with:

``` js
function requireCultures(callback) {
    var culturePaths = [];
    // ...
    require(culturePaths, callback);
}
```

``` js
require([translationPath], _.bind(function(translation) {
    // ...
}, this));
```

In the first example, I have only an identifier, which I trust to be a function callback. In the second, I am using Underscore.js and I can affirm that this expression will return a function (and it even takes a `FunctionExpression` I might otherwise have passed directly to `require`), but the argument is a `CallExpression` in the AST, not a `FunctionExpression`.

I think this part of the rule should be similar to the rest and should flag _against_ `ObjectExpression`, arrays, or basic literals, but should assume that identifiers or `CallExpression` nodes could resolve to a function callback.
